fix: pass through server port when configured

cshawaus opened the pull request:
As of v9, the `port` no longer gets passed into the server. Thus, Node will default to picking the first available port. This fix restores the previous behaviour in v8 and earlier, while ensuring that a random port can be assigned when no user-defined port is provided.

eduardoboucas commented:
@cshawaus Thanks for the fix and sorry for the slow response on this! It's been released in https://github.com/netlify/primitives/releases/tag/blobs-v9.1.2.

I'm curious, though: what's your use case for the `BlobsServer` class? We've been looking at it mostly as an internal package, so I'd love to hear more about how you're using it.

cshawaus commented:
@eduardoboucas Thank you for the update. We use the package to emulate Netlify platform features in a completely agnostic and predictable way. Using the `concurrently` package, we spin up a Blobs server alongside the Next.js dev server, which allows us to have 100% parity when developing features built on top of Blobs.

We found this far more efficient than relying on the Netlify CLI, which was limited by the number of available seats, as not every developer had a seat.

Below is the actual script we use. I added a bit of flair to handle some signals, which dramatically improved the shutdown experience.

```js
import { BlobsServer } from '@netlify/blobs/server'

const BLOBS_PORT = parseInt(process.env.PORT ?? '3000', 10) + 10
const DEBUG_STRINGS = process.env.WEB_DEBUG?.split(',') ?? []

// Create a server by providing a local directory where all
// blobs and metadata should be persisted
const server = new BlobsServer({
  debug: DEBUG_STRINGS.includes('blobs'),
  directory: '.blobs/cache',
  port: BLOBS_PORT,
  token: 'super-secret-blobs-token',
})

// Read more about the BlobsServer options here:
// https://github.com/netlify/blobs?tab=readme-ov-file#server-api-reference
await server.start()

console.debug(`Blobs development server is running on http://localhost:${BLOBS_PORT}`)

// Stop the server when the process is terminated
for (const signal of ['SIGINT', 'SIGTERM', 'SIGQUIT']) {
  process.on(signal, async () => {
    console.log()

    // Force kill after 5 seconds
    const killTimer = setTimeout(() => {
      console.error(
        'Blobs development server could not be stopped in time... forcefully exiting',
      )
      process.exit(1)
    }, 5000)

    await server.stop()

    clearTimeout(killTimer)

    console.log('Blobs development server has been stopped')
    process.exit(0)
  })
}
```
